pkg/api/plugin.go

```diff
@@ -14,12 +14,17 @@ const (
 	ContextKeyTenantID     ContextKey = "TenantID"
 )
 
+// DeployFn makes it possible to plug in different logic to the deploy.
```

Comment:
Provide suitable documentation here that is aimed at the implementor.  The implementor must initiate a deployment of the given template using mode resources.Incremental and wait for it to complete.